lib/src/rdb/datastore.rs

```diff
@@ -397,10 +373,39 @@ impl RocksdbDatastore {
     ///
     /// # Arguments
     /// * `path`: The file path to the rocksdb database.
-    /// * `max_open_files`: The maximum number of open files to have. If
-    ///   `None`, the default will be used.
-    pub fn new_db<P: AsRef<Path>>(path: P, max_open_files: Option<i32>) -> Result<Database<RocksdbDatastore>> {
-        let opts = get_options(max_open_files);
+    pub fn new_db<P: AsRef<Path>>(path: P) -> Result<Database<RocksdbDatastore>> {
+        let opts = RocksdbDatastore::get_options(None);
+        let path = path.as_ref();
+
+        let db = match DB::open_cf(&opts, path, CF_NAMES) {
+            Ok(db) => db,
+            Err(_) => {
+                let mut db = DB::open(&opts, path)?;
+
+                for cf_name in &CF_NAMES {
+                    db.create_cf(cf_name, &opts)?;
+                }
+
+                db
+            }
+        };
+
+        let metadata_manager = MetadataManager::new(&db);
+        let indexed_properties = metadata_manager.get_indexed_properties()?;
+        drop(metadata_manager);
+
+        Ok(Database::new(RocksdbDatastore {
+            db: Arc::new(db),
+            indexed_properties: Arc::new(RwLock::new(indexed_properties)),
+        }))
+    }
+
+    /// Creates a new rocksdb datastore with user-tuned rocksdb Option.
+    ///
+    /// # Arguments
+    /// * `path`: The file path to the rocksdb database.
+    /// * `opts`: The user-tuned rocksdb options.
+    pub fn new_db_with_options<P: AsRef<Path>>(path: P, opts: &Options) -> Result<Database<RocksdbDatastore>> {
         let path = path.as_ref();
 
         let db = match DB::open_cf(&opts, path, CF_NAMES) {
```

Comment:
A few unnecessary refs on this line and below

Reply:
Reasons for `opts: &Options`:
1. From the view of coding, just follow the format of `DB::open_cf` etc
2. From the view of practicing, after `new_db` with `opts` the user have to clone the `opts` to run [`Options::get_statistic`](https://docs.rs/rocksdb/latest/rocksdb/struct.Options.html#method.get_statistics) because `DB` is taking the ownership of `opts`. 

As for the unnecessary refs inside, I will cleanup after fixing fuzz test